### PR TITLE
Issue 90 - Cover tile with same behavior of content type

### PR DESCRIPTION
**Pull request by rodfersou**
closes https://github.com/collective/collective.polls/issues/90

### CODE REVIEW

File: src/collective/polls/content/poll_templates/view.pt [hunk 1]
```diff
@@ -1,69 +1,79 @@
-<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en"
```
Comment (rodfersou) [inline src/collective/polls/content/poll_templates/view.pt]: Just reindented this code.

File: src/collective/polls/tiles/poll.pt [hunk 2]
```diff
@@ -1,42 +1,72 @@
...
+                  value=""
+                  tal:attributes="value view/poll_uid" />
+              <input type="submit"
+                  name="poll.submit"
+                  value="Submit"
+                  i18n:attributes="value" />
+            </form>
+          </tal:vote>
+
+          <tal:results tal:condition="python:results and not can_vote">
```
Comment (rodfersou) [inline src/collective/polls/tiles/poll.pt]: This block is new, to show results

File: src/collective/polls/static/js/collective.poll.js [hunk 3]
```diff
@@ -2,6 +2,39 @@
...
+                        var updated_info = $(info).find('[data-tile="'+$tile.attr('data-tile')+'"]');
+                        $tile.html(updated_info.html());
+                        $('.poll-data, .votePortlet form').drawpoll();
+                    }
+                });
+            };
+
+            $('.poll-tile form').on('submit', function(e) {
+                e.preventDefault();
+                var $form = $(this);
```
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: With javascript I stop the submit and do it with ajax.

File: src/collective/polls/static/js/collective.poll.js [hunk 4]
```diff
@@ -2,6 +2,39 @@
...
+                $.ajax({
+                   type: 'POST',
+                   url: $form.attr('action'),
+                   data: $form.serialize() + '&poll.submit=Submit',
+                   context: this,
+                   success: function(data) {
+                       var $form = $(this);
+                       var $tile = $form.parents('[data-tile]');
```
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: After ajax submit is done update tile.

File: src/collective/polls/static/js/collective.poll.js [hunk 5]
```diff
@@ -2,6 +2,39 @@
 
     "use strict"; // jshint ;_;
 
+    $(function() {
+        if ($('.poll-tile').length > 0) {
+            var update_tile = function($tile) {
```
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: Get page with ajax

File: src/collective/polls/static/js/collective.poll.js [hunk 6]
```diff
@@ -2,6 +2,39 @@
...
+                $.ajax({
+                    url: location.href,
+                    context: $tile,
+                    success: function(info) {
+                        var $tile = this;
+                        var updated_info = $(info).find('[data-tile="'+$tile.attr('data-tile')+'"]');
```
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: Filter just the current tile data.

File: src/collective/polls/static/js/collective.poll.js [hunk 7]
```diff
@@ -2,6 +2,39 @@
...
+                        $tile.html(updated_info.html());
+                        $('.poll-data, .votePortlet form').drawpoll();
```
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: Update html and draw the poll

File: src/collective/polls/static/js/collective.poll.js [hunk 8]
```diff
@@ -2,6 +2,39 @@
...
+                });
+            };
+
+            $('.poll-tile form').on('submit', function(e) {
+                e.preventDefault();
+                var $form = $(this);
+                $.ajax({
+                   type: 'POST',
+                   url: $form.attr('action'),
+                   data: $form.serialize() + '&poll.submit=Submit',
```
Comment (hvelarde) [inline src/collective/polls/static/js/collective.poll.js]: don't you have to escape the `&`?

**Comment by hvelarde** [comment]
@rodfersou I think it should be a good idea to add some basic tests if possible.

File: src/collective/polls/static/js/collective.poll.js (hunk 8 above)
Comment (rodfersou) [inline src/collective/polls/static/js/collective.poll.js]: not in this case.. the [$form.serialize()](https://api.jquery.com/serialize/) returns something like:
```
'single=Single&multiple=Multiple&multiple=Multiple3&check=check2&radio=radio1'
```

File: src/collective/polls/tests/test_poll.robot [hunk 9]
```diff
@@ -61,6 +64,38 @@ Test portlet poll
...
+    # drag&drop a Poll 
+    Open Content Chooser
+    Click Element  link=Content tree
+    Click Element  css=.ui-draggable a[href*=test-folder]
+    Drag And Drop  css=${poll_selector}  css=${tile_selector}
+    Wait Until Page Contains Element  css=div.poll-tile.vote-container
+
+    # move to the default view and check tile persisted
+    Click Link  link=View
+    Page Should Contain Image  css=div.poll-tile.vote-container
```
Comment (rodfersou) [inline src/collective/polls/tests/test_poll.robot]: This test is breaking here (line 96)
It looks like the tile is not added into page.. need help to figure out what is wrong here.

After this part the test should make a vote and ensure that the results are displayed.

**Comment by rodfersou** [comment]
@hvelarde test done.  I just didn't understand why tests passed for plone 4.2.. in my machine it breaks.. :)

File: src/collective/polls/content/poll_templates/view.pt (hunk 1 above)
Comment (hvelarde) [inline src/collective/polls/content/poll_templates/view.pt]: please don't do that, because it makes my life harder.

Comment (rodfersou) [inline src/collective/polls/content/poll_templates/view.pt]: ok, let's keep this ugly messed up template forever so..

**Comment by rodfersou** [comment]
@hvelarde please review this

**Pull request merged**